packages/ckeditor5-ui/src/dialog/dialog.ts

```diff
@@ -0,0 +1,354 @@
+/**
+ * @license Copyright (c) 2003-2023, CKSource Holding sp. z o.o. All rights reserved.
+ * For licensing, see LICENSE.md or https://ckeditor.com/legal/ckeditor-oss-license
+ */
+
+/**
+ * @module ui/dialog/dialog
+ */
+
+import type View from '../view.js';
+import { type Editor, Plugin } from '@ckeditor/ckeditor5-core';
+import DialogView, { type DialogViewCloseEvent, DialogViewPosition } from './dialogview.js';
+import type { DialogActionButtonDefinition } from './dialogactionsview.js';
+import type { DocumentChangeEvent } from '@ckeditor/ckeditor5-engine';
+
+/**
+ * The dialog controller class. It is used to show and hide the {@link module:ui/dialog/dialogview~DialogView}.
+ */
+export default class Dialog extends Plugin {
+	/**
+	 * The name of the currently visible dialog view instance.
+	 *
+	 * @observable
+	 */
+	declare public id: string | null;
+
+	/**
+	 * The currently visible dialog view instance.
+	 */
+	public view?: DialogView;
+
+	/**
+	 * The dialog plugin instance controlling the currently visible dialog view.
+	 *
+	 * Only one dialog can be visible at once, even if there are many editor instances on the page.
+	 * If one editor wants to show a dialog, it should first check if there is no other visible dialog already.
+	 * But only the plugin that showed the dialog should be able to hide it
+	 * as it stores the {@link #_onHide()} callback and the proper editor reference.
```

Comment:
```suggestion
	/**
	 * The `Dialog` plugin instance which most recently showed the dialog.
	 *
	 * Only one dialog can be visible at once, even if there are many editor instances on the page.
	 * If an editor wants to show a dialog, it should first hide the dialog that is already opened.
	 * But only the `Dialog` instance that showed the dialog is able able to properly hide it.
	 * This is why we need to store it in a globally available space (static property).
	 * This way every `Dialog` plugin in every editor is able to correctly close any open dialog window.
```

Reply:
As I was re-writing this docs, I had three observations:

*   This property should be private, it is an implementation detail and there should be no need for external code to reach for `visibleDialogPlugin`.
*   The name is not the best but I don't have a better proposal for now and since it is private, I wouldn't spend much time on it.
*   And most important: `Dialog#hide()` should also use `Dialog.visibleDialogPlugin`. This way I can easily close all dialogs without thinking how many editors I have on page. Honestly, using `Dialog#hide()` would be incorrect if it just operated on "itself". Especially in our own plugins, where we don't know in what editor configuration they will be used. We would have to always go for `Dialog.visibleDialogPlugin.hide()`. So, let's have it baked in.